src/workerd/api/hyperdrive.c++

```diff
@@ -0,0 +1,107 @@
+#include "hyperdrive.h"
+#include <openssl/rand.h>
+#include <cstdint>
+#include <kj/compat/http.h>
+#include <kj/encoding.h>
+#include "sockets.h"
+#include "global-scope.h"
+#include <kj/string.h>
+#include <workerd/util/uuid.h>
+
+namespace workerd::api {
+Hyperdrive::Hyperdrive(uint clientIndex, kj::String database,
+                       kj::String user, kj::String password)
+    : clientIndex(clientIndex), database(kj::mv(database)),
+      user(kj::mv(user)), password(kj::mv(password)) {
+        kj::byte randomBytes[16];
+        KJ_ASSERT(RAND_bytes(randomBytes, sizeof(randomBytes)) == 1);
+        randomHost = kj::str(kj::encodeHex(randomBytes), ".hyperdrive.local");
+      }
+
+jsg::Ref<Socket> Hyperdrive::connect(jsg::Lock& js) {
+  auto connPromise = connectToDb();
+  
+  auto paf = kj::newPromiseAndFulfiller<kj::Maybe<kj::Exception>>();
+  auto conn = kj::newPromisedStream(connPromise.then(
+      [&f = *paf.fulfiller](kj::Own<kj::AsyncIoStream> stream) {
+    f.fulfill(kj::none);
+    return kj::mv(stream);
+  }, [&f = *paf.fulfiller](kj::Exception e) {
+    KJ_LOG(WARNING, "failed to connect to local hyperdrive process", e);
```

Comment:
This log doesn't make much sense in this new context; we aren't connecting to a local hyperdrive process.

Reply:
Was it intentional that this was left as is?

Reply:
My bad, missed it